### DISCUSSION TITLE
how to improve ecimal precision of message_stats.publish_details.rate or web UI

### DISCUSSION THREAD

**Discussion by ChenjiaxingXJTU** [rabbitmq/rabbitmq-server#4834] [Questions] [answered]
when I Execute a command
```bash
rabbitmqadmin list queues name message_stats.publish_details.rate
```

the result echo x.y, (e.g., 0.2)

so I wonder how to Improve decimal precision, e.g. publish_details.rate=0.0323.

at present, if qps of publish some rabbitmq message is very low (e.g., 0.0323), the CLI or web manager UI will return 0.0

but to see https://www.rabbitmq.com/management-cli.html, publish rate example of official website offer is high precision (e.g., 100.985821591)

**Comment by mkuratczyk** [comment, marked as answer]
Hi,

I'd recommend using the Prometheus plugin to get the total and then you can
divide it by whatever time window you want, with whatever precision you
want:

$ curl -s http://localhost:15692/metrics/per-object | grep
^rabbitmq_queue_messages_published_total
rabbitmq_queue_messages_published_total{channel="<0.962.0>",queue_vhost="/",queue="q1",exchange_vhost="/",exchange="direct"}
15

Best,

On Wed, May 18, 2022 at 3:14 AM 笼鹰 ***@***.***> wrote:

> when I Execute a command
>
> rabbitmqadmin list queues name message_stats.publish_details.rate
>
> the result echo x.y, (e.g., 0.2)
>
> so I wonder how to Improve decimal precision, e.g.
> publish_details.rate=0.0323.
>
> at present, if qps of publish some rabbitmq message is very low (e.g.,
> 0.0323), the CLI or web manager UI will return 0.0
>
> but to see https://www.rabbitmq.com/management-cli.html, publish rate
> example of official website offer is high precision (e.g., 100.985821591)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/discussions/4834>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACI7PIRH6CCGTOLK3KAHXYDVKRABDANCNFSM5WGWCRKA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Michał
RabbitMQ team
